core/unit_test/TestViewCtorProp.hpp

```diff
@@ -0,0 +1,145 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+// Check Kokkos::Impl::is_view_label.
+TEST(TEST_CATEGORY, is_view_label) {
+  static_assert(Kokkos::Impl::is_view_label<std::string>::value);
+
+  constexpr unsigned N = 3;
+  static_assert(Kokkos::Impl::is_view_label<const char[N]>::value);
+  static_assert(Kokkos::Impl::is_view_label<char[N]>::value);
+
+  // A char* is not a label. Thus, a label is distinguished from a pointer type.
+  static_assert(!Kokkos::Impl::is_view_label<char*>::value);
+}
+
+// Check traits of Kokkos::Impl::ViewCtorProp<>.
+TEST(TEST_CATEGORY, vcp_empty_traits) {
+  using vcp_empty_t = Kokkos::Impl::ViewCtorProp<>;
+
+  // Check that the empty view constructor properties class is default
+  // constructible. This is needed for calls of Kokkos::view_alloc().
+  static_assert(std::is_default_constructible_v<vcp_empty_t>);
+
+  static_assert(std::is_same_v<decltype(Kokkos::view_alloc()), vcp_empty_t>);
+}
```

Comment:
What is the goal of that test?

Reply:
I guess checking that `vcp_empty_t` is default constructible is a must, because this is not straightforward (and is needed).

Checking that `Kokkos::view_alloc()` returns a `vcp_empty` is somewhat a check of the internal and could be removed. However, `is_default_constructible_v` checks that the default constructor exists, but it does not instantiate it. At some point, we had a working version that was telling "yeah it is default constructible", but any call site that would actually instantiate the default constructor would not compile. So I guess this second part of the test is about this: check that instantiating the default constructor works ?

Reply:
There are several use cases where `view_alloc` is called without arguments in `Kokkos`. And so this test ensures that such use case works.  @romintomasett's comment explains well with more details why it was needed.